### PR TITLE
[4.1.0 - Migration] Add data migrator and update scripts for 3.1.0 to 3.2.0 migration step

### PR DESCRIPTION
**Pull request by RakhithaRR**
## Purpose
This PR,
- Adds DBDataMigrator to update the API_TYPE of the APIs during 3.1.0 to 3.2.0 migration step.
- Fixes issues in mssql pre migration scripts

## Goals
The API_TYPE column is only used to distinguish APIs and APIProducts in APIM versions before 3.2.0. This migrator will update the API_TYPE of the APIs correctly in the database by checking the type attribute available in the registry artifacts.

### CODE REVIEW

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V320Migration.java [hunk 1]
```diff
@@ -22,10 +22,7 @@
 import org.wso2.carbon.apimgt.migration.migrator.VersionMigrator;
 import org.wso2.carbon.apimgt.migration.migrator.commonMigrators.PreDBScriptMigrator;
 import org.wso2.carbon.apimgt.migration.migrator.commonMigrators.RegistryResourceMigrator;
-import org.wso2.carbon.apimgt.migration.migrator.v320.IdentityScopeMigrator;
-import org.wso2.carbon.apimgt.migration.migrator.v320.SPMigrator;
-import org.wso2.carbon.apimgt.migration.migrator.v320.ScopeMigrator;
-import org.wso2.carbon.apimgt.migration.migrator.v320.V320RegistryResourceMigrator;
+import org.wso2.carbon.apimgt.migration.migrator.v320.*;
```
Comment (CrowleyRajapakse) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V320Migration.java:25]: shall we fix star import?

Comment (RakhithaRR) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V320Migration.java]: Fixed

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
+package org.wso2.carbon.apimgt.migration.migrator.v320;
```
Comment (HiranyaKavishani) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java:1]: License header?

**Review by HiranyaKavishani — APPROVED** [review]

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java (hunk 2 above)
Comment (RakhithaRR) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java:17]: Updated

**Pull request merged**